yt_idefix/data_structures.py

```diff
@@ -162,13 +169,17 @@ def _icoords_to_fcoords(
 
 
 class IdefixVtkHierarchy(IdefixHierarchy):
+    _io_type = vtk_io
+
     def _get_field_offset_index(self) -> dict[str, int]:
         return self.ds._field_offset_index
 
     @cached_property
     def _cell_widths(self) -> tuple[XSpans, YSpans, ZSpans]:
         with open(self.index_filename, "rb") as fh:
-            cell_edges = vtk_io.read_grid_coordinates(fh, geometry=self.ds.geometry)
+            cell_edges = self._io_type.read_grid_coordinates(
```

Comment:
I noticed `vtk_io.read_grid_coordinates` accepts `BinaryIO` as the first argument while `h5_io.read_grid_coordinates` accepts str or PathLike. That means you cannot generalize this implement for now. 

I believe it will raise error when `_cell_widths` and `_cell_centers` below are called.

Reply:
> I noticed `vtk_io.read_grid_coordinates` accepts `BinaryIO` as the first argument while `h5_io.read_grid_coordinates` accepts str or PathLike. That means you cannot generalize this implement for now.
> 
> I believe it will raise error when `_cell_widths` and `_cell_centers` below are called.

This is way for now I copied most of the code. This isn't ideal but works out for initial testing.

Reply:
Yeah! What's happening with the local test is that I think all the test are not running like the type check when I give `pytest`.

Reply:
> This is why for now I copied most of the code. This isn't ideal but works out for initial testing.

So everyone is on the same page: as much as I'd like this PR to be "ideal" (minimal diff, no duplicated code ...) I understand the process is easier done progressively, so my only real requirements is that the code works and is tested (we're getting there), and that the code be expressive or at least, minimally commented, to leave room for me to take the wheel and refactor after the fact.

I know the current structure of the code isn't ideal because I never intentionally designed any space for this patch to fill, but if possible I would like to take care of cleaning the rough edges myself rather than having this PR contain large structural changes.